src/vsr/superblock_free_set_encoded.zig

```diff
@@ -299,68 +305,64 @@ pub fn FreeSetEncodedType(comptime Storage: type) type {
             defer assert(set.callback == .none);
 
             assert(set.block_index == 0);
-            assert(!set.grid.?.superblock.free_set.opened);
+            assert(!set.grid.?.free_set.opened);
```

Comment:
These path changes are annoying (it seems like free set actually used to be in the grid?) and make me with we had a dependency-injection-like setup where logical things like Grid, SuperBlock, Manifest, etc, are just separately allocated in the heap somewhere, and, whenever an X needs access to Y, we just have `y: *Y` field in X, which is set up early during startup, allowing circular references and what not. Perhaps this is a wish better left unfulfilled :-) At minimum, it'll break `@fieldParentPtr` pattern, requiring more explicit contexts for callbacks.